### PR TITLE
Add support form emulated<N> arch

### LINKED ISSUE #998: scalar implementation without code duplication
I'm currently a bit confused about the purpose of `xsimd::generic`. My expectation was that its possible to use this architecture as a fallback or for development purposes.  For this I would assume that `xsimd::batch<T, xsimd::generic>` just contains a single scalar value. But this is not the case - Instead the build fails because `xsimd::has_simd_register` is `false`. 

I could specialize `load_*` and `broadcast` calls and try to separate it from the rest of the implementation which then hopefully works with scalar and simd types. But it feels like this could be abstracted through implementation of the generic architecture. 

Is there a reason why this is not done? Are there examples how to write simd and scalar implementations without duplicating code?

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Fix #998

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@JohanMabille getting close to an end with this one, but I still need to double check if it works with Pythran as it touches xsimd_scalar.

**Comment by serge-sans-paille** [comment]
pythran check ok, @JohanMabille I let you decide

File: test/test_api.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <random>
 
 #include "test_utils.hpp"
+#if 0
```
Comment (JohanMabille) [inline test/test_api.cpp:18]: Are the tests broken? Or is this a leftover?

Comment (serge-sans-paille) [reply test/test_api.cpp]: a left over

**Comment by serge-sans-paille** [comment]
@JohanMabille all green even with the tests un-commented \o/

**Pull request merged**